File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.ts
function removeDuplicates(nums: number[]): number {
    let map = new Map();
    let k = 0;

    for (let i = 0; i < nums.length; i++) {
        if (!map.has(nums[i])) {
            map.set(nums[i], true);
            nums[k] = nums[i];
            k++;
        }
    }
    
    return k;
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        dup_ptr = 0
        for j in range(len(nums)):
            if nums[dup_ptr] == nums[j]:
                continue
            else:
                dup_ptr+=1
                nums[dup_ptr],nums[j] = nums[j],nums[dup_ptr]
        return dup_ptr +1